Jellyfin.Plugin.CoverArtArchive/CoverArtArchiveImageProvider.cs

```diff
@@ -0,0 +1,141 @@
+using MediaBrowser.Model.Dto;
+using MediaBrowser.Model.Entities;
+using MediaBrowser.Model.Providers;
+using MetaBrainz.MusicBrainz.CoverArt;
+using Microsoft.Extensions.Logging;
+
+namespace Jellyfin.Plugin.CoverArtArchive
+{
+    /// <summary>
+    /// The cover art archive image provider.
+    /// </summary>
+    public class CoverArtArchiveImageProvider : IRemoteImageProvider
+    {
+        private readonly ILogger<CoverArtArchiveImageProvider> _logger;
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        private readonly CoverArt _coverArtClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CoverArtArchiveImageProvider"/> class.
+        /// </summary>
+        /// <param name="httpClientFactory">Instance of the <see cref="IHttpClientFactory"/> interface.</param>
+        /// <param name="logger">Instance of the <see cref="ILogger{CoverArtArchiveImageProvider}"/> interface.</param>
+        public CoverArtArchiveImageProvider(
+            IHttpClientFactory httpClientFactory,
+            ILogger<CoverArtArchiveImageProvider> logger)
+        {
+            _httpClientFactory = httpClientFactory;
+            _logger = logger;
+
+            _coverArtClient = new CoverArt("Jellyfin Cover Art Archive Plugin", Assembly.GetExecutingAssembly().GetName().Version!.ToString(), "apps@jellyfin.org");
+        }
+
+        /// <inheritdoc />
+        public string Name
+            => "Cover Art Archive";
+
+        /// <inheritdoc />
+        public bool Supports(BaseItem item)
+            => item is MusicAlbum;
+
+        /// <inheritdoc />
+        public IEnumerable<ImageType> GetSupportedImages(BaseItem item)
+        {
+            return new[]
+            {
+                ImageType.Primary
+            };
+        }
+
+        /// <inheritdoc />
+        public async Task<HttpResponseMessage> GetImageResponse(string url, CancellationToken cancellationToken)
+        {
+            // TODO: Should probably also use the MusicBrainz library for this.
+            _logger.LogDebug("Getting image from {Url}", url);
+            var httpClient = _httpClientFactory.CreateClient(NamedClient.Default);
+            return await httpClient.GetAsync(new Uri(url), cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc />
+        public async Task<IEnumerable<RemoteImageInfo>> GetImages(BaseItem item, CancellationToken cancellationToken)
+        {
+            var list = new List<RemoteImageInfo>();
+            var musicBrainzId = item.GetProviderId(MetadataProvider.MusicBrainzAlbum);
+
+            if (!string.IsNullOrEmpty(musicBrainzId))
+            {
+                try
+                {
+                    var release = await _coverArtClient.FetchReleaseAsync(Guid.Parse(musicBrainzId)).ConfigureAwait(false);
+
+                    var frontImage = release.Images.SingleOrDefault(image => image.Types == CoverArtType.Front);
```

Comment:
```suggestion
                    var frontImage = release.Images.FirstOrDefault(image => image.Types == CoverArtType.Front);
```

Just a reminder that `SingleOrDefault` will throw an exception if multiple matches are found, and it fully iterates through the list.